service/azureconfig/v1/resource/vnetpeering/current.go

```diff
@@ -0,0 +1,35 @@
+package vnetpeering
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/azure-operator/client"
+	"github.com/giantswarm/azure-operator/service/azureconfig/v1/key"
+	"github.com/giantswarm/microerror"
+)
+
+// GetCurrentState retrieve the current host cluster virtual network peering resource from azure.
+func (r Resource) GetCurrentState(ctx context.Context, azureConfig interface{}) (interface{}, error) {
+	a, err := key.ToCustomObject(azureConfig)
+	if err != nil {
+		return nil, microerror.Maskf(err, "GetCurrentState")
+	}
+
+	return r.getCurrentState(ctx, a)
+}
+
+func (r Resource) getCurrentState(ctx context.Context, azureConfig providerv1alpha1.AzureConfig) (interface{}, error) {
```

Comment:
Idea of having corresponding private method is to pass strong type parameter and return strong typed values. Here you return an interface `interface{}`. It should be `network.VirtualNetworkPeering{}`. There is aleadyt mess with nil values, which returning strong typed values prevents.